src/foam/dao/AbstractDAO.java

```diff
@@ -58,6 +58,8 @@ protected Sink decorateSink_(Sink sink, long skip, long limit, Comparator order,
       sink = new PredicatedSink(predicate, sink);
     }
 
+    ((AbstractFObject) sink).setX(x);
```

Comment:
This is mainly to stay consistent with the other methods within the abstract class which accept a context. Ex. `pipe`, `removeAll_`.